extensions/parquet/base/src/main/java/io/deephaven/parquet/base/util/CachedChannelProvider.java

```diff
@@ -196,12 +216,20 @@ public boolean isOpen() {
         public void close() throws IOException {
             Require.eqTrue(isOpen, "isOpen");
             isOpen = false;
+            clearContext();
             returnPoolableChannel(this);
         }
 
         private void dispose() throws IOException {
             wrappedChannel.close();
         }
+
+        @Override
+        public void setContext(@NotNull final SeekableChannelContext channelContext) {
```

Comment:
You can't use `@NotNull` here, since you expect to be able to clear a context by setting it to `null`.
```suggestion
        public void setContext(final SeekableChannelContext channelContext) {
```